client/quota.go

```diff
@@ -58,6 +58,7 @@ type QuotaJournalValues struct {
 	Size       quantity.Size `json:"size,omitempty"`
 	RateCount  int           `json:"rate-count,omitempty"`
 	RatePeriod time.Duration `json:"rate-period,omitempty"`
+	RateValid  bool          `json:"rate-valid,omitempty"`
```

Comment:
`RateSet` or `RateValid`? RateValid feels semantically better, but I have a feeling that we tend to use `Set` around in the project instead

Reply:
we don't have anything else like this, we need to think a bit

Reply:
Alternative approach:

I could wrap it in a struct like the other values instead of having this bool indicating whether or not the values are set

Reply:
yes, but you don't need to add another level of nesting though, you can do something like:

```
type QuotaJournalRate struct {
        RateCount  int           `json:"rate-count"`  // no omitempty
	RatePeriod time.Duration `json:"rate-period,omitempty"`  
}

type QuotaJournalValues struct {
   ...
   *QuotaJournalRate   // embedded pointer
}
```

on the unmarshaling side you can check of .QuotaJournalRate is nil or not.  On the client side fill in with a non-nil pointer if the value should be considered set.

Here's play ground to show what is involved: https://go.dev/play/p/Qkxxja-Ln4a

Reply:
Thanks for the suggestion! It was a much better way of solving it. I've updated the PR with this approach and also verified the new approach works as expected.